deps/src/singular.cpp

```diff
@@ -5,6 +5,25 @@
 #include "ideals.h"
 #include "matrices.h"
 
+std::string singular_return;
+std::string singular_error;
+std::string singular_warning;
```

Comment:
Are any of these meant to be used in other C++ files? If not, they should be static. Also, does this project have any naming conventions for global variables? If not, perhaps that should be considered?

Reply:
Will make them static. Afaik, we have no naming convention for globals, neither variables nor functions.